languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/style.xml

```diff
@@ -9914,6 +9914,49 @@ USA
         </rulegroup>
 
 
+        <rule id='FAZER_EXERCER' name="[Universitário][Científico] Verbo Fazer → Verbo Exercer" type="style" tone_tags="academic" is_goal_specific="true" tags="picky" default="temp_off">
+            <antipattern>
+                <token postag='V.+' postag_regexp='yes'/>
+                <token min='0' max='1' postag='D[AI].+|NC.+|AQ.+' postag_regexp='yes'/>
+                <token inflected='yes'>atividade</token>
+                <token inflected='yes'>físico</token>
+                <example>Conclusão, estava em jejum e fazendo atividade física pesada.</example>
+                <example>Faça atividades físicas todos os dias.</example>
+                <example>Tom nunca faz atividades físicas.</example>
+            </antipattern>
+            <antipattern>
+                <token postag='V.+' postag_regexp='yes'/>
+                <token min='0' max='1' postag='D[AI].+|NC.+|AQ.+' postag_regexp='yes'/>
+                <token inflected='yes'>força</token>
+                <token regexp='yes' inflected='yes'>&languages;|aéreo|armada|espacial|marítimo|militar|naval|terrestre</token>
+                <example>Com o objetivo de fazer as forças alemãs acreditarem que a invasão aliada viria de Pas-de-Calais, a força fantasma foi esta...</example>
+                <example>O país vai fazer as forças terrestres deslocarem-se.</example>
+                <example>É isso que fazem as forças espaciais.</example>
+            </antipattern>
```

Comment:
This antipattern has nothing really to do with military forces, that's completely epiphenomenal to what you're trying to counter. The structure that prevents 'fazer' from being replaceable with 'exercer' in this context is the **causative** use of 'fazer'.

- _queremos **fazer** essa profissão **pagar** melhor_
- _o colestoral **faz** a pressão **subir**_
- _eles dizem que tais filmes **fazem** a violência **tornar-se** mais popular_

I think we'd do better to just block all matches of `fazer $NP $verb-INF`.

Reply:
> This antipattern has nothing really to do with military forces, that's completely epiphenomenal to what you're trying to counter. The structure that prevents 'fazer' from being replaceable with 'exercer' in this context is the **causative** use of 'fazer'.
> 
>     * _queremos **fazer** essa profissão **pagar** melhor_
> 
>     * _o colestoral **faz** a pressão **subir**_
> 
>     * _eles dizem que tais filmes **fazem** a violência **tornar-se** mais popular_
> 
> 
> I think we'd do better to just block all matches of `fazer $NP $verb-INF`.

@p-goulart 

What is “`$NP`”?

Thanks!

Reply:
Ahhhhh… they are the valid words, if I use `NC.+` or `NC.+|AQ.+` it will cover them all.

Reply:
ahhhh... I will start generating the results against a 900 000 sentences database from zero with the new tokens, and then enable the APs one by one.
`java -Dfile.encoding=UTF-8 -Xmx4500M -jar languagetool-wikipedia.jar check-data -l pt-PT -r FAZER_EXERCER -f pt-BR.txt --max-sentences 900000 --context-size 100 >__0new.txt`

Reply:
Sorry, NP here was short for 'noun phrase'.